server/routes/companies.js

```diff
@@ -116,25 +179,459 @@ router.get('/filter', function(req, res, next) {
+      let length = event_ids.length;
+      for (let n=0; n < length; n++) {
+        companyEvents.push({
+           company_id: company_id,
+           event_id: event_ids[n]
+        })
+      }
+    } else {
+        companyEvents = {
+           company_id: company_id,
+           event_id: event_ids
+        }
+    }
+    var queryEvent = knex('event_company').insert(companyEvents)
+  }
+
+  //remove
+  let removeUsers = []
+  if (remove_user_ids) {
+    if (Array.isArray(remove_user_ids)) {
+      let length = remove_user_ids.length;
+      for (let n=0; n < length; n++) {
+        removeUsers.push(remove_user_ids[n]);
+      }
+    } else {
+      removeUsers.push(remove_user_ids);
+    }
+  }
+  let removeQueryRank = knex('user_company_rank').whereIn('user_id', removeUsers).andWhere("company_id", company_id).del()
+  
+  let removePositions = []
+  if (remove_position_ids) {
+    let length = remove_position_ids.length;
+    if(Array.isArray(remove_position_ids)){
+      for (let n=0; n < length; n++) {
+        removePositions.push(remove_position_ids[n]);
+      }
+    } else {
+      removePositions.push(remove_position_ids);
+    }
+  }
+  let removeQueryPos = knex('company_position').whereIn('position_id', removePositions).andWhere("company_id", company_id).del()
+
+  let removeMajors = []
+  if (remove_major_ids) {
+    let length = remove_major_ids.length;
+    if(Array.isArray(remove_major_ids)){
+      for (let n=0; n < length; n++) {
+        removeMajors.push(remove_major_ids[n]);
+      }
+    } else {
+      removeMajors.push(remove_major_ids);
+    }
+  }
+  let removeQueryMajor = knex('company_major').whereIn('major_id', removeMajors).andWhere("company_id", company_id).del()
+
+  let removeContacts = []
+  if (remove_contact_ids) {
+    let length = remove_contact_ids.length;
+    if(Array.isArray(remove_contact_ids)){
+      for (let n=0; n < length; n++) {
+        removeContacts.push(remove_contact_ids[n]);
+      }
+    } else {
+      removeContacts.push(remove_contact_ids)
+    }
+  }
+  let removeQueryContact = knex('company_contact').whereIn("contact_id", removeContacts).andWhere("company_id", company_id).del()
+
+  let removeEvents = []
+  if (remove_event_ids) {
+    let length = remove_event_ids.length;
+    if(Array.isArray(remove_event_ids)){
+      for (let n=0; n < length; n++) {
+        removeEvents.push(remove_event_ids[n])
+      }
+    } else {
+      removeEvents.push(remove_event_ids)
+    }
+  } 
+  let removeQueryEvent = knex('event_company').whereIn("event_id", removeEvents).andWhere("company_id", company_id).del()
+
+  var queryCompany = knex('company').update(values).where({ id: req.params.company_id })
+  knex.transaction(async function(trx) {
+    //modify company attributes
+    if (req.body.name || req.body.website || req.body.logo || req.body.citizenship_requirement || req.body.description) {
+      await queryCompany;
+    }
+
+    //insert
+    if (user_ids && ranks) {
```

Comment:
one-liner these: 
if (user_ids && ranks) await queryRank.transacting(trx);

https://stackoverflow.com/questions/8860654/javascript-single-line-if-statement-best-syntax-this-alternative